unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Video/DCLVideoTexture.cs

```diff
@@ -156,9 +158,18 @@ public override IEnumerator ApplyChanges(BaseModel newModel)
 
         private void Initialize(DCLVideoClip dclVideoClip)
         {
+            if (isInitialized) return;
+            isInitialized = true;
+
             string videoId = (!string.IsNullOrEmpty(scene.sceneData.id)) ? scene.sceneData.id + id : scene.GetHashCode().ToString() + id;
             texturePlayer = new WebVideoPlayer(videoId, dclVideoClip.GetUrl(), dclVideoClip.isStream, videoPluginWrapperBuilder.Invoke());
             texturePlayerUpdateRoutine = CoroutineStarter.Start(OnUpdate());
+            // unsub first
+            CommonScriptableObjects.playerCoords.OnChange -= OnPlayerCoordsChanged;
+            CommonScriptableObjects.sceneID.OnChange -= OnSceneIDChanged;
+            scene.OnEntityRemoved -= SetPlayStateDirty;
+            Settings.i.audioSettings.OnChanged -= OnAudioSettingsChanged;
+            // then resub
```

Comment:
This subscribe/unsubscribes in one place doesn't fill correct for me. As I understand, this `Initialize` method will be called only once, means that it will not be any subscription before. Also, since unsubscription happens in Dispose, maybe it highlights the problem with the disposal of the object. Did you try to remove this part and check if it works the same?

Reply:
done